source/val/validate.h

```diff
@@ -112,7 +112,7 @@ spv_result_t ModuleLayoutPass(ValidationState_t& _, const Instruction* inst);
 spv_result_t CfgPass(ValidationState_t& _, const Instruction* inst);
 
 /// Performs Id and SSA validation of a module
-spv_result_t IdPass(ValidationState_t& _, const spv_parsed_instruction_t* inst);
+spv_result_t IdPass(ValidationState_t& _, Instruction* inst);
```

Comment:
Note, non-const because when we call RegisterInstruction it needs a non-const instruction to add to all_definitions. Changing all_definitions to const .... ripples ....